File: 2-arrays.py
numbers = [1, 2, 3, 4, 5]

print(numbers[0])
print(numbers[-1])
print(len(numbers))

doubled = [number * 2 for number in numbers]
even_numbers = [number for number in numbers if number % 2 == 0]
total = sum(numbers)

print(doubled)
print(even_numbers)
print(total)

numbers.append(6)
print(numbers)

File: 2-arrays.rb
# Ruby array 配合 Enumerable 方法，寫 map/select/sum 會好順手。
numbers = [1, 2, 3, 4, 5]

puts numbers.first
puts numbers.last
puts numbers.length

doubled = numbers.map { |number| number * 2 }
even_numbers = numbers.select(&:even?)
sum = numbers.sum

puts doubled.inspect
puts even_numbers.inspect
puts sum

numbers << 6
puts numbers.inspect
